### PR TITLE
add / to the allowlist filter when normalizing the top-namespace

### PR DESCRIPTION
**Pull request by ryaminal**
## Description
Adding the forward slash `/` to the filter for top-namespace when normalizing

## Motivation and Context

We would like to create a top level with something like `foo/bar` so that the module can be imported as `import foo.bar.stuff`.

## How Has This Been Tested?

Tested this locally on my mac. Should work on most *nix systems, but I'm not certain how it will impact Windows.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] I have read the [code of conduct](https://github.com/davidvujic/poetry-multiproject-plugin/blob/master/CODE-OF-CONDUCT.md).
- [X] I have read the [contributing guide](https://github.com/davidvujic/poetry-multiproject-plugin/blob/master/CONTRIBUTING.md).
- [X] I have updated the documentation accordingly (if applicable).

### CODE REVIEW

**Comment by ryaminal** [comment]
It's possible this is a breaking change for some folks that use `--with-top-namespace` with a slash already. before this change, the slash is just replaced with an empty string so a namespace of `foo/bar` would have just been normalized to `foobar`. This change would certainly break some folks if they were using `/`.

**Comment by DavidVujic** [comment]
Thank you @ryaminal for this Pull Request!

It seems very reasonable and useful, I'll have a look!

File: README.md [hunk 1]
```diff
@@ -59,33 +59,46 @@ poetry build-project --with-top-namespace my_namespace
...
     /my_package
        __init__.py
        my_module.py
 ```
 
+###### Namespace with path(`--with-top-namespace=my_namespace/subdir`)
+```shell
+my_namespace/
+    /my_package
+        /subdir
```
Comment (DavidVujic) [inline README.md:81]: minor: I think the directories should be:
my_namespace/
     /subdir
         /my_package

**Review by DavidVujic — APPROVED** [review]
👏 ⭐

File: README.md (hunk 1 above)
Comment (DavidVujic) [reply README.md:81]: I'll merge your pull request, and will update the README separately - no worries!

**Comment by DavidVujic** [comment]
Thank you for highlighting the possible breaking change! I think it could be a good idea to bump the major version of this plugin, to 2.0.0. I'll do this in a separate pull request.

**Review by DavidVujic — COMMENTED** [review]
@ryaminal 

I'm sorry! I didn't test this out properly before merging, and had to revert the merge.

When I test this out:
```
poetry build-project --directory projects/my_fastapi_project --with-top-namespace=hello/world
```

The result is invalid Python:
```python
from hello/world.demo import message
from hello/world.demo.log import get_logger
```

Do you want to have another look at it and re-open, or create a new Pull Request?

**Comment by DavidVujic** [comment]
The result of `normalize_top_namespace` is a string that is expected to be a proper Python namespace, in addition to when building up a OS directory path.

Maybe it should be two different functions:
one for normalizing a Python namespace, and another one for normalizing a directory path?

**Comment by ryaminal** [comment]
Oh crap, you're right. The few files I checked when I was testing never actually imported anything in the top_namespace so i never saw the problem. Amateur hour. I'll work on this a bit today and get a better solution.

Sorry to waste your time.

**Comment by ryaminal** [comment]
Replace this PR with #34 where the problems surfaced here are handled. Made sure to test it a little more thoroughly.

**Pull request merged**